src/services/expenses.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+let expenses = [];
+
+function getFiltered(data) {
+  const { userId, categories, from, to } = data;
+
+  if (userId) {
+    expenses = expenses.filter(expense => +userId === expense.userId);
+  }
+
+  if (categories) {
+    expenses = expenses.filter(expense => expense.category === categories);
+  }
+
+  if (from && to) {
+    expenses = expenses.filter(({ spentAt }) => (
+      spentAt >= from && spentAt <= to
+    ));
+  }
+
+  return expenses;
+}
+
+function getById(expenseId) {
+  const foundExpense = expenses.find((expense) => expense.id === +expenseId);
+
+  return foundExpense || null;
+}
+
+function create(data) {
+  const newExpense = {
+    id: Math.random(expenses.length + 1),
```

Comment:
why `Math.random`?
Use `uuid` lib or make a separate function where you find max id & make a plus one to this value
Reuse this!